composer.json

```diff
@@ -13,8 +13,8 @@
         }
     },
     "replace": {
-        "guzzlehttp/guzzle": "*",
-        "guzzlehttp/psr7": "*"
+        "guzzlehttp/guzzle": "^7.5",
+        "guzzlehttp/psr7": "^2.4"
```

Comment:
I thought that the `"*"` in the "replace" section allowed the guzzle to come from core somehow.

PR #1007 updated `composer.json` but did not commit the resulting `composer.lock`. It seems that composer is now reporting this problem (I guess that there was a newer version of composer released.

Can we get it to pass without changing this guzzle stuff?

Reply:
@phil-davis i will look into it.

Reply:
@ phill it passes with old php digest `"image": "owncloudci/php:7.4@sha256:3ec4e600382f88ede77c5fb6a012d63c8a469d805420a0eb5044d7bc91c8da18",`. since the new php update it seems failing.

Reply:
gives a problem when updating the composer
 `Problem 1
    - owncloud/files_external_s3 is present at version dev-master and cannot be modified by Composer
    - roave/security-advisories dev-latest conflicts with guzzlehttp/psr7 <1.8.4|>=2,<2.1.1 (owncloud/files_external_s3 dev-master replaces guzzlehttp/psr7 *).
    - Root composer.json requires roave/security-advisories dev-latest -> satisfiable by roave/security-advisories[dev-latest].
`

Reply:
And the `roave/security-advisories` package conflicts with guzzle other than `7.5` and psr7 other than `2.4`. for this composer update is needed @phil-davis ?

Reply:
I think that the `replace` section actually declares that this app/package `files_primary_s3` can "replace" those guzzle dependencies - the `"*"` means that it correctly "replaces" any guzzle version. I think that this is a bit weird - normally this "replace" thing would be used when the repo is actually a fork of the thing that it replaces, and the fork contains all the code of the original plus bug-fixes and extra features, so the fork can really directly do everything that the original repo did (and better).

But in this case, I think, we are saying that `files_primary_s3` when it is running will have working guzzle code already available to it "somewhere" - and that "somewhere" is inside `owncloud/core` which will have some useful/working guzzle loaded. We use the "replace" section just to tell composer that it does not need to load the code of those guzzle dependencies, because the needed code will already be there.

And there is "roave/security-advisories" - that has some restrictions on various guzzle repo versions, because of old bugs. It wants to ensure that there will be a "good enough" guzzle available. And it gets concerned about `"*"` 

Mentioning the guzzle versions seems a reasonable thing to do - the guzzle that is provided in core matches these version constraints. So I will approve and merge this PR.

I anyone really knows about this composer replace stuff in a use-case like this, then feel free to explain it!